bitcoin/src/lib.rs

```diff
@@ -867,27 +868,47 @@ impl BitcoinCoreApi for BitcoinCore {
     /// # Arguments
     /// * `txid` - transaction ID
     /// * `num_confirmations` - how many confirmations we need to wait for
+    /// * `block_hash` - optional block hash
     async fn wait_for_transaction_metadata(
         &self,
         txid: Txid,
         num_confirmations: u32,
+        block_hash: Option<BlockHash>,
+        is_wallet: bool,
     ) -> Result<TransactionMetadata, Error> {
-        let (block_height, block_hash, fee) = retry(get_exponential_backoff(), || async {
-            Ok(match self.rpc.get_transaction(&txid, None) {
-                Ok(GetTransactionResult {
-                    info:
-                        WalletTxInfo {
-                            confirmations,
-                            blockhash: Some(hash),
-                            blockheight: Some(height),
-                            ..
-                        },
-                    fee,
-                    ..
-                }) if confirmations >= 0 && confirmations as u32 >= num_confirmations => Ok((height, hash, fee)),
-                Ok(_) => Err(Error::ConfirmationError),
-                Err(e) => Err(e.into()),
-            }?)
+        let (block_hash, fee) = retry(get_exponential_backoff(), || async {
+            if is_wallet {
```

Comment:
Can we use https://github.com/interlay/interbtc-clients/blob/eeb2fda6d584dbf851ad851dc013e3cf52463de4/bitcoin/src/lib.rs#L750C27-L750C27 ? I'm not sure it works, it looks like it might need decoding, but would be nice to avoid having to branch here

Reply:
We would need to manually calculate the fee for the metrics which I'd like to avoid adding right now (to unblock the next release).